velox/expression/tests/CastExprTest.cpp

```diff
@@ -798,6 +797,15 @@ TEST_F(CastExprTest, primitiveValidCornerCases) {
   setCastIntByTruncate(true);
   // To integer.
   {
+    // Valid strings.
+    testCast<std::string, int8_t>("tinyint", {"1.2"}, {1}, false);
+    testCast<std::string, int8_t>("tinyint", {"-1.8"}, {-1}, false);
+    testCast<std::string, int8_t>("tinyint", {"1."}, {1}, false);
```

Comment:
cases from L803-L807 are very unexpected and not obvious. Can you mention that these are valid cases as well in. the description for `cast_to_int_by_truncate`